### DISCUSSION TITLE
What is an example of using jest.mock("ky") with mockResolvedValue?

### DISCUSSION THREAD

**Discussion by srmxlt** [sindresorhus/ky#488] [Q&A] [answered]
import ky from "ky";

jest.mock("ky");

 it("should fetch data", () => {
    ky.post.mockResolvedValue(() => ({
       json: () => Promise.resolve(resp),
    }));
    }
  )

the snippet above results in an error
"_ky.default.post(...).json is not a function"

The main question is: What is the correct way to use jests' mockResolvedValue with ky? I have not found any examples of this usage yet. any examples would be greatly appreciated.
Thank you!

**Comment by jobdn** [comment]
Hey! Did you solve this problem?

**Comment by VijethKC** [comment, marked as answer]
main.js

```highlight
const data= await ky.post('https://www.example.com').json()
```

main.test.js

```highlight
// import ky
import ky from 'ky'

// mock ky instance
jest.mock('ky', () => ({
  post: jest.fn(),
  json: jest.fn(),
}))

// Test block
describe('Ky Tests', () => {
  it('renders component', async () => {
    const mockResponse = {}

   // Spy ky here
    jest.spyOn(ky, 'post').mockReturnThis()
    jest.spyOn(ky, 'json').mockReturnValue(mockResponse)

    render(
        <Component/>
    )

    await screen.findByRole('')
    expect(1).toBe(1)
  })
})
```
